sqlglot/dialects/duckdb.py

```diff
@@ -96,6 +97,21 @@ def _json_format_sql(self: generator.Generator, expression: exp.JSONFormat) -> s
     return f"CAST({sql} AS TEXT)"
 
 
+def _ensure_utf8(charset: exp.Literal) -> None:
+    if charset.name.lower() != "utf-8":
+        raise UnsupportedError(f"Unsupported charset {charset}")
```

Comment:
this is not quite correct, you should call self.unspported on the generator

Reply:
Done